src/api/donations.py

```
@@ -0,0 +1,659 @@
+            sub_id,
+            getattr(row, "stripe_customer_id", None),
+            invoice_pi or None,
+        ).run()
+
+        print(f"[webhook] Recurring donation {new_id} recorded for subscription {sub_id}")
+        await _send_thank_you_email(env, {
+            "email":  getattr(row, "donor_email", "") or "",
+            "amount": getattr(row, "amount", 0),
+            "type":   "monthly",
+        })
+    except Exception as exc:
```

Ruff (0.16.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)